framework/src/kernels/MatReaction.C

```diff
@@ -30,21 +33,30 @@ MatReaction::MatReaction(const InputParameters & parameters)
     _v_name(_is_coupled ? coupledName("v") : _var.name()),
     _v(_is_coupled ? coupledValue("v") : _u),
     _v_var(_is_coupled ? coupled("v") : _var.number()),
-    _L(getMaterialProperty<Real>("mob_name")),
+    _L(isParamValid("mob_name") ? getMaterialProperty<Real>("mob_name")
+                                : getMaterialProperty<Real>("reaction_rate")),
```

Comment:
```suggestion
    _L(getMaterialProperty<Real>("mob_name")),
```

Reply:
```suggestion
    _L(getMaterialProperty<Real>("reaction_rate")),
```
is better because you won't have to update this line when you remove the deprecated parameter